### ISSUE TITLE
Fix alaska scraper to return approximate dates

### ISSUE THREAD

**Issue by mlissner** [freelawproject/juriscraper#170]
Alaska has a frustrating website:

http://www.courtrecords.alaska.gov/webdocs/opinions/sp.htm

The problem is that rather than putting the items in the website with actual dates, they do quarterly date spans like:

> July - September 2016

These used to be the exception, I believe, but now they're the norm. So...we need to deal with them.

In https://github.com/freelawproject/courtlistener/issues/533 we're adding a new field to the DB for approximate dates. We should add support for this to Juriscraper as well. I think the process would be:

 - [ ] Update Juriscraper to have a new function, something like, `get_date_truthiness` (<-- That's probably bad). The new function would return True for every date that was known to be accurate, and would return False for ones that are estimates.

 - [ ] Update Alaska to return this value as appropriate. I suppose some of the time they used to provide accurate dates?

 - [x] Update CL to have the needed field in the DB and to show it in appropriate places, maybe with an FAQ and a (?) thing.

 - [x] Update the code that calls Juriscraper in CL so that it sets the values in the DB as we scrape.

**Comment by arderyp** [comment]
Sweet, thanks for the info and explination.  This sounds like a good plan.  I'm a little unclear on the `get_date_truthiness()` function.  Are you envisioning this as a new method on the AbstractSite that defaults to: `retrun [True] * len(self.cases)`

Then it would be up to us on states like alaska to build the appropriate list of True/False values by overriding/implemening `get_date_truthiness()`.

As for approximating the middle-most date in these specific Alaska date ranges, I think we could probably write some lib function to do that.  I haven't delved that deeply into it, but it looks like it might be somewhat complex, and require building a list of all dates in the range.  I don't know yet.  The simpler solution would be for us to say that we are just going to use some arbitrary date in the range.  Assuming the example, "July - September 2016", some candidates are:

- July 1 [start of range/quarter]
- September 1 [middle-ish date that is easy to determine without need for complex looping or datetime calls]
- September 30th [end of range/quarter]

**Comment by mlissner** [comment]
Yeah, the thing you described for `get_date_truthiness` is exactly what I had in mind.

For the date span thing, yeah, hard coding it rather than computing the middle date seems fine to me. I guess one easy optimization could be to do (pseudocode):

   estimate = parse("July - September 2016")
   max(today, estimate)

That'd make sure that we never had a date in the future, and in general—if our scrapers are running every day, and if they post things promptly—we'd actually have the correct date.

Watch out when you write this that they're going to be jerks and use every different kind of dash under the sun: `-`,`–`, `­—` (minus, n-dash, m-dash).

**Comment by arderyp** [comment]
good point with the dashes.  I was palnning to get around that with split(), something like:
date_string = "July - September 2016"

```
try:
    date = convert_date_string(date_string)
except ValueError:
    parts = date_string.split()
    if len(parts) == 4:
        estimate_string = '%s 1, %s' % (parts[0], parts[3]) # July 1, 2016
        estimate_date = convert_date_string(estimate_string)
        date = max(today, estimate_date)
    else:
        raise InsanityException("Unrecognized date format")
```

Does that looks reasonable enough?

**Comment by mlissner** [comment]
Assuming it's got a test or two, yeah, that looks good to me.

**Comment by arderyp** [comment]
right, good idea

**Comment by mlissner** [comment]
I've done the work for this on CourtListener. There's a new field in the DB called `date_filed_is_approximate` (a bit wordy, but it's explicit). For this to work, the function in `OpinionSite` should be called `_get_date_filed_is_approximate` and it should return `True` or `False` for all results from all scrapers.

**Comment by arderyp** [comment]
Do you need me to implement that on AbstractSite?  If so, is it a simple matter of retuning True * len(self.cases), or do I need to write code to handle database interactions too?

**Comment by mlissner** [comment]
`date_filed` is only on the `OpinionSite`, not on the other objects that override `AbstractSite`, so we'll want to put it there, not on `AbstrictSite`, where it would get inherited by oral arguments and other things in the future.

The other thing that needs to happen is that the new field needs to be added to the `_req_attrs` attribute on `OpinionSite`. By adding it there we make sure that it gets processed by the `parse` method.

I think the return `[False] * len(self.case_names)` should work. I'm not sure if doing `* len(self.cases)` would work. Maybe?

The DB side of things is all taken care of in CL as of a moment ago.

**Comment by arderyp** [comment]
Okay, I'll take a look after work
